Allow different encodings for reading JCAMP files

andreasdoll opened the pull request:
Allow to provide a list of possible encodings for reading JCAMP files,
and use the first encoding which doesn't raise an UnicodeDecodeError.
Suppress such errors, except when all provided encodings fail.
Keep the current behaviour by choosing the default encoding if no
encoding is specified.

I'm not sure if you want to pull this, but for my purposes this is easier and more futureproof than converting the JCAMP files on operating system level.

jjhelmus commented on nmrglue/fileio/bruker.py:
```diff
@@ -2091,7 +2092,7 @@ def rm_dig_filter(
 # JCAMP-DX functions
 
-def read_jcamp(filename):
+def read_jcamp(filename, encodings=[locale.getpreferredencoding()]):
```
I think it would be preferable to use a single encoding here.  If users want to try multiple encoding they can create a loop over `read_jcamp` with a check for a UnicodeDecodeError.

jjhelmus commented on nmrglue/fileio/bruker.py:
```diff
@@ -2121,28 +2122,36 @@ def read_jcamp(filename):
-                    warn("Unable to correctly parse line:" + line)
+    for N, encoding in enumerate(encodings):
+        try:
+            with open(filename, 'r', encoding = encoding) as f:
```
In Python 2 `encoding` is not an argument in the buildin `open` command.  `io.open` does support this argument.

jjhelmus commented:
@andreasdoll I made two suggestions on how I think this should be implemented.  Let me know if you need any help with this.

andreasdoll commented:
Thanks @jjhelmus, I wasn't aware that encodings in open() is not backwards compatible. I've modified that commit according to your wishes and added the previously forgotten parameter docstring.

jjhelmus commented:
LGTM, thanks @andreasdoll
